include/dynd/callable.hpp

```diff
@@ -155,9 +155,9 @@ namespace nd {
 
     std::intptr_t get_narg() const { return get_type()->get_npos(); }
 
-    const ndt::type &get_arg_type(std::intptr_t i) const { return get_type()->get_pos_type(i); }
+    const ndt::type get_arg_type(std::intptr_t i) const { return get_type()->get_pos_type(i); }
 
-    const array &get_arg_types() const { return get_type()->get_pos_types(); }
+    nd::array get_arg_types() const { return get_type()->get_pos_types(); }
```

Comment:
Yes, I'll try.  In general there are still many conversions from std::vector -> nd::array because all the consumers expect an nd::array.  Can these be eliminated in the end?